src/sensor/ping.h

```diff
@@ -648,5 +648,5 @@ class Ping : public Sensor
         int waiting = 0;
     };
     friend QDebug operator<<(QDebug d, const Ping::messageStatus& other);
-    QHash<PingPing1DNamespace::ping_ping1D_id, messageStatus> requestedIds;
+    QHash<uint16_t, messageStatus> requestedIds;
```

Comment:
This is a high level class, it's not save to use uint16_t as key, use enum.
The enum is explicit and allows the compiler to detect problems.

Reply:
a uint16_t is safe by definition - a message id is a uint16_t. To change the signature here will also require a single enum scope for all message ids (common, ping1d, ping360), which we do not have (we could make it if we require reserved ids for each sensor).

You can still gain the compiler/enum advantages: I recommend the caller make use of the the individual message id enum/namespaces.

Reply:
From the creator of C++ language:

> enum class -- scoped and strongly typed enums
The enum classes ("new enums", "strong enums") address three problems with traditional C++ enumerations:
conventional enums implicitly convert to int, causing errors when someone does not want an enumeration to act as an integer.
conventional enums export their enumerators to the surrounding scope, causing name clashes.
the underlying type of an enum cannot be specified, causing confusion, compatibility problems, and makes forward declaration impossible.

http://www.stroustrup.com/C++11FAQ.html#enum

There is also the cpp guidelines:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#enum-enumerations

Reply:
Thank you, I've read it and I understand the enum, the enum class, type safety and the advantages of using them. I still think it is the wrong idea to have an enum type for the parameter. (But I think it is the right idea for the caller to always use enum types for the arguments! ie `request(Ping1dId::DISTANCE)`. You can still write your code using enum types in every call.

Because you can, does not mean you should. Because it compiles, does not make it right. It is similar to our past conversation about if we should declare something const here:  https://github.com/bluerobotics/ping-protocol/pull/77